Allow gossip header for reply-to address

azul opened the pull request:
In order to enable encrypted replies in the presence of a Reply-To header
we allow specifying Autocrypt-Gossip for the given address.

azul commented:
Thinking about this...
What happens if I receive an encrypted mail with a Reply-To header right now?
Will a reply create an unencrypted message to that address?
Is there a warning?

hpk42 commented on doc/level1.rst:
```diff
@@ -682,12 +685,15 @@ this case contains one Autocrypt-Gossip header for each recipient,
   referenced by ``addr``. See also :ref:`preliminary recommendation`
   for how this key is selected.
+  For ``Reply-To`` headers it MUST contain the public key
+  which should be used to encrypt a reply.
```
what if a sender has the keys for To/CC recipients (which it must in order to encrypt) but not for Reply-To?

Valodim replied:
this is per individual header, so in that case there simply wouldn't be a header with an `addr` to match the `Reply-To`

dkg replied:
I'm not convinced that MUST is particularly meaningful in this new sentence -- how could it be contradicted?  I'm also pedantically concerned that it seems to imply that any reply (even those that aren't addressed to the `Reply-To` address) should be encrypted to that key.

dkg commented on doc/level1.rst:
```diff
@@ -682,12 +685,15 @@ this case contains one Autocrypt-Gossip header for each recipient,
 
 - MUST include an ``addr`` attribute that matches one of the
-  recipients in the ``To`` or ``Cc`` headers.
+  recipients in the ``To``, ``Cc``, or ``Reply-To`` headers.
```
I think this should change "recipients" to "addresses", since the address in a `Reply-To` header is not a recipient.

Valodim commented on doc/level1.rst:
```diff
@@ -672,6 +672,9 @@ attribute indicates the recipient address this header is valid for as
 See example in :ref:`autocrypt-gossip-example`
 
+The ``Autocrypt-Gossip`` header can also be used
+to include keys for the address specified in the ``Reply-To`` header.
```
It's pedantic, but we might want to mention that this is only useful in cases where that address differs from the ones in From/To/Cc

Valodim commented on doc/level1.rst:
```diff
@@ -710,8 +716,8 @@ Autocrypt peer state of the gossiped recipient identified by its
 1. If ``gossip-addr`` does not match any recipient in the mail's
-   ``To`` or ``Cc`` header, the update process terminates (i.e.,
-   header is ignored).
+   ``To``, ``Cc``, or ``Reply-To`` header,
```
same as above, "recipient" should now be "address"

azul commented:
Updated the PR:
* talking about `addresses` rather than `recipients`
  because Reply-To headers contain an address but not a recipient.
* using SHOULD rather than MUST for picking the key for
  Reply-To gossip
* explaining Reply-To gossip makes sense if the address is not
  in To or Cc headers already.
* Dropped the requirement for the message to have multiple recipients.
  Gossip may make sense if only one recipient but also a Reply-To header
  are present.
